jsconfig.json

```diff
@@ -0,0 +1,5 @@
+{
+  "compilerOptions": {
+    "checkJs": true
+  }
```

Comment:
Probably needs `"target": "es2015"` otherwise I see a lot of errors in VSCode. And it seems that VSCode ignores `.eslintrc` for which I don't know a proper fix.

Reply:
`es2018`, actually, because this is what's specified already in eslintrc.

Reply:
Having `.eslintrc` i.e. without extension is deprecated *([see](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-file-formats))*
That's why I renamed it to `.eslintrc.yml`

---

Yes, I have changed it to `es2018`.

Reply:
I still see errors about GM_ functions being undefined though. Not really a problem since I don't use VSCode but maybe you can fix it here?

Reply:
Here is what I came up with the new commit:
- downloaded the typing files (index.d.ts) from https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/greasemonkey/v3
and https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/greasemonkey, 
- moved them (together with the types.d.ts) into a separate folder.
- linked them in jsconfig.json

Reply:
Better but it still doesn't use my eslintrc so VSCode shows tons of errors over the entire script...

Reply:
Are you sure? With my PR I have renamed `.eslintrc` to `.eslintrc.yml`
and it works ok:
<details>
<summary><code>eslint --debug script.user.js</code></summary>

<!-- leave a blank line above -->
![2021-04-05_170648](https://user-images.githubusercontent.com/723651/113582693-5d6e3680-9631-11eb-9c79-acc247295b26.jpg)
</details>

Reply:
VSCode, not cli. One of the advantages of using an IDE is that it can perform eslint checks as you type right in the source editor.

Reply:
No, in VSCode it shows also just 1 error for me:

![2021-04-05_172824](https://user-images.githubusercontent.com/723651/113585123-77f5df00-9634-11eb-8b32-f494e25e9adb.jpg)

The rest 138 errors are all TypeScript errors.

Reply:
I gave the cli screenshot above to show that the .eslintrc.yml is indeed the config that is being used.

Reply:
Ah well, maybe it's a bug in my version of VSCode or the eslint extension. Doesn't matter to me though so I guess the PR is good.

Reply:
I've merged this, thank you.

Reply:
Turns out VSCode needs these two settings to disable its own "smart" (not) checker:

```
    "typescript.validate.enable": false,
    "javascript.validate.enable": false,
```

Reply:
Thanks for the info, I just made a PR having them in VSCode's Workspace settings, instead of disabling them globally.